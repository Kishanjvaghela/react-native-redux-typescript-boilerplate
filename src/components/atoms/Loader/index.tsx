import * as React from 'react';
import {
  Text,
  View,
  ActivityIndicator,
  ActivityIndicatorProps,
} from 'react-native';
import styles from './styles';

export default class Loader extends React.Component<ActivityIndicatorProps> {
  render() {
    return (
      <View style={styles.root}>
        <ActivityIndicator />
        <Text style={styles.loaderTitle}>Please wait ...</Text>
      </View>
    );
  }
}
