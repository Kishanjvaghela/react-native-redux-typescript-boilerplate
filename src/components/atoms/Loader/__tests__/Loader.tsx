import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../index';

it('renders correctly with defaults', () => {
    const button = renderer
        .create(<Loader />)
        .toJSON();
    expect(button).toMatchSnapshot();
});