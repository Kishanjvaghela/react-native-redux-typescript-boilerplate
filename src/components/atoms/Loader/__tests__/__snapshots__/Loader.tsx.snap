// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    Please wait ...
  </Text>
</View>
`;
